### PR TITLE
some issues from 200-300 + code security

### PR DESCRIPTION
**Pull request by livingrockrises**

### CODE REVIEW

File: contracts/smart-contract-wallet/SmartAccount.sol [hunk 1]
```diff
@@ -68,11 +68,11 @@ contract SmartAccount is
     
     // Events
     // EOA + Version tracking
-    event ImplementationUpdated(address indexed _scw, string version, address newImplementation);
+    event ImplementationUpdated(address indexed _scw, string indexed version, address indexed newImplementation);
```
Comment (AmanRaj1608) [inline contracts/smart-contract-wallet/SmartAccount.sol:74]: As per discussion do we need this many indexed filters? @filmakarov suggested this as gas optimisation to have less indexed parameter.

File: package.json [hunk 2]
```diff
@@ -58,8 +58,8 @@
     "@ethersproject/abstract-signer": "^5.6.2",
     "@ethersproject/constants": "^5.6.1",
     "@nomiclabs/hardhat-etherscan": "^2.1.6",
-    "@openzeppelin/contracts": "^4.7.3",
-    "@openzeppelin/contracts-upgradeable": "^4.7.3",
+    "@openzeppelin/contracts": "^4.8.0",
+    "@openzeppelin/contracts-upgradeable": "^4.8.0",
```
Comment (AmanRaj1608) [inline package.json:62]: We can use the https://github.com/OpenZeppelin/openzeppelin-contracts/releases/tag/v4.8.1 which replaces the call to staticcall

My this PR will update it later.
https://github.com/bcnmy/scw-contracts/pull/40/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519

File: contracts/smart-contract-wallet/SmartAccount.sol [hunk 3]
```diff
@@ -253,6 +253,7 @@ contract SmartAccount is
         address gasToken,
         address payable refundReceiver
     ) private returns (uint256 payment) {
+        require(tokenGasPriceFactor != 0, "invalid stokenGasPriceFactor");
```
Comment (AmanRaj1608) [inline contracts/smart-contract-wallet/SmartAccount.sol:294]: is this typo `stokenGasPriceFactor` or am i missing something?

**Review by AmanRaj1608 — APPROVED** [review]
LGTM

File: contracts/smart-contract-wallet/SmartAccount.sol (hunk 1 above)
Comment (livingrockrises) [reply contracts/smart-contract-wallet/SmartAccount.sol:74]: hmm I am keeping 3 indexed for every event. post merge we can check all events and optimise what we need to index

File: contracts/smart-contract-wallet/SmartAccount.sol (hunk 3 above)
Comment (livingrockrises) [reply contracts/smart-contract-wallet/SmartAccount.sol:294]: typo..fixing it

Comment (livingrockrises) [reply contracts/smart-contract-wallet/SmartAccount.sol]: done

**Pull request merged**